cpu/stm32/periph/gpio_all.c

```diff
@@ -49,19 +49,27 @@ static gpio_isr_ctx_t isr_ctx[EXTI_NUMOF];
 #define EXTI_REG_FTSR       (EXTI->FTSR1)
 #define EXTI_REG_PR         (EXTI->PR1)
 #define EXTI_REG_IMR        (EXTI->IMR1)
+#define EXTI_CR             SYSCFG
+#elif defined(CPU_FAM_STM32MP1)
+#define EXTI_REG_RTSR       (EXTI->RTSR1)
+#define EXTI_REG_FTSR       (EXTI->FTSR1)
+#define EXTI_REG_PR         (EXTI->PR1)
+#define EXTI_REG_IMR        (EXTI_C2->IMR1)
+#define EXTI_CR             EXTI
 #else
 #define EXTI_REG_RTSR       (EXTI->RTSR)
 #define EXTI_REG_FTSR       (EXTI->FTSR)
 #define EXTI_REG_PR         (EXTI->PR)
 #define EXTI_REG_IMR        (EXTI->IMR)
+#define EXTI_CR             SYSCFG
 #endif
 
 /**
  * @brief   Extract the port base address from the given pin identifier
  */
 static inline GPIO_TypeDef *_port(gpio_t pin)
 {
-    return (GPIO_TypeDef *)(pin & ~(0x0f));
+    return (GPIO_TypeDef *)(pin & ~(0x0fff));
```

Comment:
I guess other family members should still use the old mask.

Btw.: Why is the wider mask needed?
Can there now be more GPIOs / Port?

Reply:
You are right, the new mask should not be a replacement of the mask for other families.
But as we want the port base address, for other stm32 families, the mask should be ~(0x3FF) as the port number start at the 10th bits.
For stm32mp1, port number is at the 12th bit thus the mask is ~(0xFFF).
It seems more "logical" to me to erase the whole unwanted part even if it should "never" exceed 0xF.
As there is no functional change, do you prefer me to keep the old syntax ?